hpe/dl360/exporter.go

```diff
@@ -370,3 +509,47 @@ func (e *Exporter) exportMemoryMetrics(body []byte) error {
 
 	return nil
 }
+
+// The getDriveEndpoint function is used in a recursive fashion to get the body response
+// of any type of drive, NVMe, Physical DiskDrives, or Logical Drives, using the GenericDrive struct
+// This is used to find the final drive endpoints to append to the task pool for final scraping.
+func getDriveEndpoint(url, host string, client *retryablehttp.Client) (GenericDrive, error) {
+	var drive GenericDrive
+	var resp *http.Response
+	var err error
+	retryCount := 0
+	req := common.BuildRequest(url, host)
+	resp, err = common.DoRequest(client, req)
+	if err != nil {
+		return drive, err
+	}
+	defer resp.Body.Close()
+	if !(resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices) {
+		if resp.StatusCode == http.StatusNotFound {
+			for retryCount < 3 && resp.StatusCode == http.StatusNotFound {
+				time.Sleep(client.RetryWaitMin)
+				resp, err = common.DoRequest(client, req)
+				retryCount = retryCount + 1
+			}
+			if err != nil {
+				return drive, err
+			} else if !(resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices) {
+				return drive, fmt.Errorf("HTTP status %d", resp.StatusCode)
+			}
+		} else {
+			return drive, fmt.Errorf("HTTP status %d", resp.StatusCode)
+		}
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
`ioutil` is deprecated, you should use `io.ReadAll()` and that applies for everywhere